src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/Nodes/TreeNode.cs

```diff
@@ -229,6 +229,7 @@ public NodeInfo ToNodeInfo()
                 IsLeaf = this.IsAlwaysLeaf,
                 Label = this.Label,
                 NodePath = this.GetNodePath(),
+                ParentNodePath = this.Parent?.GetNodePath() ?? "",
```

Comment:
Why not get it set up to actually have the Parent link in the NodeInfo? That seems a lot more useful

Reply:
this is designed to be consumed by ADS side.

Reply:
Sure, and it would still be useful there. Having to duplicate any parent properties in children nodes as well isn't exactly a great pattern. And then what happens if we want to walk up the tree to find the root node or something for example?

Reply:
The node path is just like the id of the node, ADS core (OE service) has a collection of all the nodes, if we need to walk up the tree, we can call getParent function (the one I modified in the ADS PR.) until it reaches the root of tree. I don't think it is a good idea to include all the ancester nodes in every node, not necessary and not efficient.

Reply:
You definitely wouldn't include the node objects themselves. You would pass a parentId or something and then on the ADS side convert it into a reference to the actual parent object. 

Although if you have a getParent method already then why wouldn't you just call node.getParent().nodePath on the ADS side?

Reply:
the getParent() needs to use the parentNodePath to find the parent node.

Reply:
Oh ok, so this is for doing what I was suggesting - you're just using the path for the "id". That makes sense then.